lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -455,26 +462,27 @@ def install_puppet_from_deb_on( host, opts )
         #
         # @note on windows, the +:ruby_arch+ host parameter can determine in addition
         # to other settings whether the 32 or 64bit install is used
-        def install_puppet_from_msi_on( host, opts )
-
-          version = opts[:version]
+        def install_puppet_from_msi_on( hosts, opts )
+          block_on hosts do |host|
+            version = opts[:version]
 
-          if version && !version_is_less(version, '4.0.0')
-            if opts[:puppet_agent_version].nil?
-              raise "You must specify the version of puppet agent you " +
-                    "want to install if you want to install Puppet 4.0 " +
-                    "or greater on Windows"
-            end
+            if version && !version_is_less(version, '4.0.0')
```

Comment:
This check may break when we move agent versioning forward to 4.y.z, as mentioned on the dev list.